m4/sage_spkg_configure.m4

```diff
@@ -185,22 +185,40 @@ AC_DEFUN([SAGE_SPKG_CONFIGURE], [
 
 # SYNOPSIS
 #
-#   SAGE_SPKG_DEPCHECK(PACKAGE-DEPENDENCIES, FURTHER-CHECK)
-#                                 $1              $2
+#   SAGE_SPKG_DEPCHECK(PACKAGE-DEPENDENCIES, FURTHER-CHECK, [FAIL= [sage_spkg_install_]SPKG_NAME=yes])
+#                                 $1              $2                $3
 #
 # DESCRIPTION
 #     *** to be called from SAGE_SPKG_CONFIGURE* ***
 #     check for space-separated list of package dependencies $1 of package SPKG_NAME
 #     do $2 if successful
 #
+#     $3 is for off-label use - checking a reverse dependency. Then the default action of
+#     setting sage_spkg_install_SPKG_NAME=yes may be overwritten, typically to [], i.e. nothing.
+#
 AC_DEFUN([SAGE_SPKG_DEPCHECK], [
     m4_foreach_w([DEP], $1, [
        AC_REQUIRE([SAGE_SPKG_CONFIGURE_]m4_toupper(DEP))])
     AC_MSG_CHECKING([whether any of $1 is installed as or will be installed as SPKG])
     AS_IF([test x = y m4_foreach_w([DEP], $1, [ -o [x$sage_spkg_install_]DEP = xyes])], [
         AC_MSG_RESULT([yes; install SPKG_NAME as well])
-        [sage_spkg_install_]SPKG_NAME=yes], [
+        m4_default($3, [sage_spkg_install_]SPKG_NAME=yes)
+	], [
         AC_MSG_RESULT([no])
         $2
         ])
 ])
+
+
+# SYNOPSIS
+#
+#   SAGE_SPKG_REVDEPCHECK(PACKAGE-REVERSE-DEPENDENCIES)
```

Comment:
And the problem is that the abstraction (using this macro) obscures why it has to be done in the REQUIRED-CHECK ($2) argument of SAGE_SPKG_CONFIGURE.

Reply:
the abstraction always obscures the boilerplate it replaces. I'll document where it should go

Reply:
One can avoid this effect for example by finding the right name for the abstraction. `SAGE_SPKG_REVDEPCHECK` does not include the word REQUIRE. Perhaps it should

Reply:
> One can avoid this effect for example by finding the right name for the abstraction. `SAGE_SPKG_REVDEPCHECK` does not include the word REQUIRE. Perhaps it should

Feel free to propose a name.

Reply:
We could rename `SAGE_DEPCHECK` to `SAGE_SPKG_INSTALL_IF_ANY_DEP_INSTALLED_ELSE`

and `SAGE_SPKG_REVDEPCHECK` to `SAGE_SPKG_UNREQUIRE_IF_NO_REVDEP_INSTALLED`

which I think are high autoconfspeak style